.github/workflows/astro_deploy.yml

```
@@ -0,0 +1,34 @@
+name: Astro deploy
+
+on:
+  push:
+    branches:
+      # Push events on main branch
+      - main
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Git checkout
+        uses: actions/checkout@v2
```

I recommend using the latest version here.
```suggestion
        uses: actions/checkout@main
```